File: rating-service/src/main/java/com/gnomeshift/ratingservice/dto/RatingEvent.java
package com.gnomeshift.ratingservice.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class RatingEvent {
    private Long userId;
    private Long courseId;
    private Integer rating;
    private String eventType = "RATING_CREATED";
    private Long timestamp;
}

File: recommendation-service/src/main/kotlin/com/gnomeshift/recommendationservice/dto/RatingEvent.kt
package com.gnomeshift.recommendationservice.dto

import kotlinx.serialization.Serializable

@Serializable
data class RatingEvent(
    val userId: Long,
    val courseId: Long,
    val rating: Int,
    val eventType: String = "RATING_CREATED",
    val timestamp: Long
)
